Extension/src/LanguageServer/client.ts

```diff
@@ -620,6 +645,36 @@ class DefaultClient implements Client {
         this.model.tagParserStatus.Value = notificationBody.status;
     }
 
+    private updateInactiveRegions(params: InactiveRegionParams): void {
+        let renderOptions: vscode.DecorationRenderOptions = {
+            light: { color: "rgba(175,175,175,1.0)" },
+            dark: { color: "rgba(155,155,155,1.0)" }
+        };
+        let decoration: vscode.TextEditorDecorationType = vscode.window.createTextEditorDecorationType(renderOptions);
+
+        // Recycle the active text decorations when we receive a new set of inactive regions
+        let valuePair: decorationRangesPair = this.inactiveRegionsDecorations.get(params.uri);
+        if (valuePair !== undefined) {
```

Comment:
Either do ```!valuePair``` or ```this.inactiveRegionsDecorations.has(params.uri)```

Reply:
`!valuePair` changes the meaning of the statement, and `this.inactiveRegionsDecorations.has(params.uri)` is (assumedly) as expensive as calling `this.inactiveRegionsDecorations.get(params.uri)` which while minor would be the best to avoid. Is that the case?

Reply:
no need to call `.has`.  You can change it to `if (valuePair)` or just leave it as `if (valuePair !== undefined)`.  I think the 'truthy' compare argument for this case has been bolstered a bit by the recent regression, but I believe the API contract here is relatively well defined.

```
interface Map<K, V> {
    clear(): void;
    delete(key: K): boolean;
    forEach(callbackfn: (value: V, key: K, map: Map<K, V>) => void, thisArg?: any): void;
    get(key: K): V | undefined;
    has(key: K): boolean;
    set(key: K, value: V): this;
    readonly size: number;
}
```

Reply:
Whoops, should have looks at the return values of get(). 🙈 

-resolved-

Reply:
I believe `if (valuePair)` is more idiomatic and as it does not change the meaning of the statement, I will substitute it in. Thanks